intel_extension_for_transformers/llm/runtime/graph/developer_document.md

```diff
@@ -0,0 +1,440 @@
+
+=====itrex result======
+she open the door and see him. She looks at him and says, "How do you do?" He says, "Fine." She says, "What do you want?" He says, "I want to go home." She says, "Where are you going?" He says, "I'm going home." She says, "Where are you going?" He says, "I'm
+```
+
+The Korean prompt would have the output like:
+```bash
+=====pytorch result======
+옛날 옛적에 어린 소녀가 있었어요. 그 소녀는 어느 날, 숲 속에서 길을 잃고 헤매다가 한 마리의 동물을 만나게 되었어요. 그 동물은 소녀에게 길을 안내해 주겠다고 하였어요. 소녀는 그 동물을 따라 숲 속으로 들어갔어요. 한참을 걷고 있는데, 갑자기 동물이 소녀를 땅 속으로 끌고 들어가는 것이었어요. 소녀는 깜짝 놀라서 소리쳤어요. "안돼! 나를 죽이려고 하는 거야?" 그러자 동물은 소녀에게 조용히 말했어요. "쉿! 조용히 해. 내 말을 잘 들어봐. 저 앞에
+
+=====itrex result======
+옛날 옛적에 어린 소녀가 있었어요. 그 소녀는 어느 날, 숲 속에서 길을 잃고 헤매다가 한 마리의 동물을 만나게 되었어요. 그 동물은 소녀에게 길을 안내해 주겠다고 하였어요. 소녀는 그 동물을 따라 숲 속으로 들어갔어요. 한참을 걷고 있는데, 갑자기 동물이 소녀를 땅 속으로 끌고 들어가는 것이었어요. 소녀는 깜짝 놀라서 소리쳤어요. "안돼! 나를 죽이려고 하는 거야?" 그러자 동물은 소녀에게 조용히 말했어요. "쉿! 조용히 해. 내 말을 잘 들어봐. 저 앞에
+```
+
+Once you make sure your model has the same generated tokens as PyTorch, you can deploy it by using python codes of `transformers` style liked and `INT4` data type. Please refer to `Python API` section for more details.
+
+# Enable graph cpp model process
+We enable a CPP model in the following four steps.
+
+<img src="imgs/Enable_cpp_model.PNG" width=1200 height=150 alt="Enable cpp model">
+<br>
+
+# 1.	Model conversion
+
+We need to implement a convert function to convert the model from pytorch format to binary, which is mainly divided into the following three steps.
+
+## 1.1.	Hyperparamters
+The term **"hyperparamters"** describes a value that is used to configure the behavior of a large language model; this is in contrast to the model's parameters, which are the weight that were derived in the training process that was used to create the model. Each model defines its own hyperparameter structure that defines the hyperparameter values accepted by that model. Valid ITREX graph files must list these values in the correct order, and each value must be represented using the correct data type. Although hyperparameters are different across models, some attributes appear in the hyperparameters for most models:
+- n_vocab: the size of the model's vocabulary
+- n_embd: the size of the model's " embedding layer", which is used during prompt ingestion.
+- n_layer: the number of layers in the model; each layer represents a set of weights.
+Here we will use [convert_gptneox.py](https://github.com/intel/intel-extension-for-transformers/blob/graph_developer_document/intel_extension_for_transformers/llm/runtime/graph/scripts/convert_gptneox.py#L96) as an example,
+```python
+fout.write(struct.pack("i", hparams["num_attention_heads"]))
+fout.write(struct.pack("i", hparams.get("n_head_kv", 0)))  # multi-query attention
+fout.write(struct.pack("i", hparams["num_hidden_layers"]))
+```
+The above `fout` is the file we need to get, and here the `num_attention`, `n_head_kv`, and `num_hidden_layer` from hparams is written into fout.
```

Comment:
and here the `num_attention`, `n_head_kv`, and `num_hidden_layer` from hparams is written into fout.
Comment 
change to
and the `num_attention`, `n_head_kv`, and `num_hidden_layer` from hparams is written into fout.
Comment